### ISSUE TITLE
hds.world.skill missing

### ISSUE THREAD

**Issue by themadfitz** [hdneo/mxo-hd#2]
Hi Alex,
Just cloned the current repo and noticed that the namespace definition for hds.world.skill is missing - is there a missing class file somehwere?  Is the code in this repo current with the binaries on the HD site (0.2 i think?).  If not will the source for that be coming up soon?

Cheers,
Hayden

**Comment by vitalyo7** [comment]
I had the same issue, likely due to some "in progress commits". Try to comment things out until errors go away, and it should work.

**Comment by themadfitz** [comment]
Thanks mate, I ended up doing that... Server starts now but it won't authenticate for love or money ;)  working through it now.

Before I go trawling through the code, is there specific format we can create the users in the database (mandatory keys etc...)?  I reversed the MD5 hashes off the default user and it still won't connect for some reason.  I'll keep digging... probably a config thing on my side.

**Comment by vitalyo7** [comment]
User is loluser/lolpass, this is from the old school menu days...

You have to check the authentication code, to see how user/pass is stored. I would recommend writing another piece of code so you can add/modify users from cmd line.

**Comment by themadfitz** [comment]
Yep, that's the plan - is suspect that Alex/Neo_HD has already written it given the sign-up form on mxoemu.info.  I'd rather just see that merged but I'll have a crack at re-writing it.  Need to understand the encryption piece as well; there are two binary blobs in the table and the salt column which I'll need to understand in more detail.

Are you still working on your code?  I'm looking through the old Singularity code as there seems to be a lot of work already done in there; regarding calculations.

**Comment by vitalyo7** [comment]
I am still working on it, but it is slow. Currently trying to clean up network + scripting code so that it may be possible to write game content using scripts.

**Comment by themadfitz** [comment]
Hmmm,
Is your build still able to be logged into?  I've recompiled a number of times now and still can't get it to auth.  The code looks right and it sends the AS_PublicKeyResponse then the client simply ACKS in and send a FIN to close the connection.  The same behaviour with Neo's live test server.  I wonder whether there's something in the ,exe that has time-bombed the client past a certain date...

**Comment by vitalyo7** [comment]
The issue is that Rajko/HardlineDreams/Singularity have their own set of private/public keys, so they will choke during authentication.

To make it work for Singularity, you need to make a copy of client folder (just so it doesn't collide with other emulator), then copy the files from src/resources/client_install/ to the root of the client. pubkey.dat is important, and you have to use singularityLauncher/bat scripts to make it work.

Also for local testing you need to edit the host file as well as clientopts.cfg as per Instructions.txt.

If you run into trouble, just ping Static Variable on the mxo discord, i can guide you thought the issues.

**Comment by themadfitz** [comment]
That did the trick - Singularity Launcher with the HDS pubkey.dat file.  I had started hacking around with the 128 byte sequence in the original matrix.exe file but this has saved me a heap of time.  I might zip up the executable and the pubkey.dat and include them in the repo.

**Comment by hdneo** [comment]
Thank you Static for helping - i was busy in the last weeks during moving location here. 
And i will be out of internet maybe for some weeks (just smartphone-only internet).

As Static mentioned you need to change the key in the matrix.exe with the correct keypair (or you have a already prepared one from client_install folder). 
I already thought about that issue that you need 3 clients installed only for playing on the 3 server types (Rajko/Singularity/HardlineDreams).

Maybe in the future it would be nice to have just one client (this would work if we have 3 different folders and add a "launcher" for it which just copy the files you want when you start the game). 
But i just had the idea and no time and prio to care about it :) 

I will check why the skill File is missing (i am not sure if it is already used somewhere).

**Comment by nexustheru** [comment]
`namespace hds.world.Skill
{
    class JumpStep
    {
  
        private LtVector3f currentJumpPos;
        private LtVector3f ltVector3f;
        private float maybeMaxHeight;
        private ushort v;
        

        public JumpStep(LtVector3f currentJumpPos, LtVector3f ltVector3f, float maybeMaxHeight, uint endTime, ushort v, uint maybeTimeBasedValue)
        {
            this.currentJumpPos = currentJumpPos;
            this.ltVector3f = ltVector3f;
            this.maybeMaxHeight = maybeMaxHeight;
            this.endTime = endTime;
            this.v = v;
            this.maybeTimeBasedValue = maybeTimeBasedValue;
        }

        public LtVector3f FromPos { get; set; }
        public  toPos ToPos { get; set; }
        public float JumpHeight { get; set; }
        public uint endTime { get; set; }
        public ushort jumpId { get; set; }
        public uint maybeTimeBasedValue { get; set; }
    }
}
`  or something similar and it compiler stops whining aswell as server..

**Comment by neowhoru** [comment]
I added some missing stuff that wasn't commited before - maybe it works now a little bit better.
The HJ is still really buggy.
